Added some project information and missing commands

czarneckid opened the pull request:
Just getting my feet wet with Clojure and Redis and noticed a number of commands not present.

Commit:
Added lein-related ignore and added some project information

Commit:
Added lein-related ignore and added some project information

Commit:
Added flush-db command

Commit:
Added flush-db command

Commit:
Added select function to allow you to select a given DB

Commit:
Added select function to allow you to select a given DB

Commit:
Added random-key function which will return a random key from the keyspace

Commit:
Added random-key function which will return a random key from the keyspace

Commit:
Added function to ask the server to close the connection

Commit:
Added function to ask the server to close the connection

Commit:
Added size function to return number of keys in currently selected database

Commit:
Added size function to return number of keys in currently selected database

Commit:
Fixing getrange function to call substr

Commit:
Fixing getrange function to call substr

Commit:
Added disconnect function

Commit:
Added disconnect function

Commit:
Added echo function

Commit:
Added echo function

mmcgrana commented:
Thanks for this pull request. Sorry I've taken forever to check it out ): I understand if you've moved on already.

I did merge the 54a4713 commit for the updated project info: On the others I have a few comments/questions:
- Do you understand the concurrency/leasing implication of connection-level commands like select, disconnect, and quit? For example, is quit supposed to quit the entire client or just the individual leased connection? Should that connection be replaced in the poll? Should select do so for the entire pool, or one connection?
- Is there any reason size isn't dbsize, like the redis command? http://redis.io/commands/dbsize
- There are some whitespace errors - can you correct those and rebase?

Thanks,
Mark

Commit:
Merge branch 'master' of github.com:czarneckid/clj-redis

Commit:
Change size to dbsize

czarneckid commented:
- quit should just affect the individual connection. I assume that connection would be returned to the pool and would re-connect, but I'd have to look at the behavior implemented there in Redis.
- select should only select a new DB for the individual connection IMO because the pool may have been initialized with a given DB. I expect it to be not common behavior for a connection to have to pull from another DB.

I've changed size to dbsize. Probably just mental block on my part :)

I will send another pull request that hopefully fixes the whitespace errors. I've rebased so hopefully all is good.

mmcgrana commented:
Do you have an example of how one would use `quit` and `select`? Given the current leasing semantics it's not clear to me what they would mean.

For the pull request, could you `rebase -i` it so that it is clearer what is being changed?
